File: BlogSiteReact16.Web/ClientApp/src/Home/BlogPost.js
import React from 'react';
import { Link } from 'react-router-dom';
import format from 'date-fns/format'

function BlogPost({ blogPost }) {
    const { id, title, text, comments, dateCreated } = blogPost;

    return <div className="card mb-4">
        <div className="card-body">
            <h2 className="card-title">{title}</h2>
            <p className="card-text">{text.legnth < 200 ? text : text.substring(0, 200)}</p>
            <small>{comments.length} comments</small>
            <br />
            <Link to={`/viewpost/${id}`} style={{ textDecoration: 'none' }}>
                <button className='btn btn-primary mt-3'>Read More &rarr;</button>
            </Link>
        </div>
        <div className="card-footer text-muted">
            Posted on {format(new Date(dateCreated), 'EEEE LLLL do, R')}
        </div>
    </div>
}

export default BlogPost;

File: BlogSiteReact16.Data/BlogPost.cs
using System;
using System.Collections.Generic;

namespace BlogSiteReact16.Data
{
    public class BlogPost
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public DateTime DateCreated { get; set; }

        public List<Comment> Comments { get; set; }
    }
}
